### PR TITLE
API for geometry

### PR DESCRIPTION
**Pull request by feilin-flexcompute**
added geometry api

### CODE REVIEW

File: flow360/component/geometry.py [hunk 1]
```diff
@@ -0,0 +1,246 @@
...
+        Get geometry from cloud
+        :param geometry_id:
+        :return:
+        """
+        return cls(geometry_id)
+
+    @classmethod
+    def from_file(
+        cls,
+        geometry_files: Union[List[str], str],
```
Comment (maciej-flexcompute) [inline flow360/component/geometry.py:112]: `file_names` would be better name here

File: flow360/component/geometry.py [hunk 2]
```diff
@@ -0,0 +1,246 @@
...
+from typing import List, Union
+
+from ..cloud.rest_api import RestApi
+from ..exceptions import Flow360FileError, Flow360ValueError
+from ..log import log
+from .interfaces import GeometryInterface
+from .resource_base import Flow360Resource, Flow360ResourceBaseModel, ResourceDraft
+from .utils import shared_account_confirm_proceed, validate_type
+
+supportedGeometryFilePatterns = [
```
Comment (maciej-flexcompute) [inline flow360/component/geometry.py:18]: use UPPER_CASE here

File: flow360/component/geometry.py [hunk 3]
```diff
@@ -0,0 +1,246 @@
...
+            if not os.path.exists(geometry_file):
+                raise Flow360FileError(f"{geometry_file} not found.")
+
+        if self.name is None and len(self.geometry_files) > 1:
+            raise Flow360ValueError(
+                "name field is required if more than one geometry files are provided."
+            )
+
+    @property
+    def geometry_files(self) -> List[str]:
```
Comment (maciej-flexcompute) [inline flow360/component/geometry.py:191]: `file_names`

File: flow360/component/geometry.py [hunk 4]
```diff
@@ -0,0 +1,246 @@
...
+        print("debug=============")
+        print(data)
+
+        if self.solver_version:
+            data["solverVersion"] = self.solver_version
+
+        resp = RestApi(GeometryInterface.endpoint).post(data)
+        info = GeometryMeta(**resp)
+        self._id = info.id
+        submitted_mesh = Geometry(self.id)
```
Comment (maciej-flexcompute) [inline flow360/component/geometry.py:233]: submitted_geometry

File: flow360/component/geometry.py (hunk 1 above)
Comment (feilin-flexcompute) [reply flow360/component/geometry.py:112]: fixed.

File: flow360/component/geometry.py (hunk 2 above)
Comment (feilin-flexcompute) [reply flow360/component/geometry.py:18]: fixed

File: flow360/component/geometry.py (hunk 3 above)
Comment (feilin-flexcompute) [reply flow360/component/geometry.py:191]: fixed. all geometry_files are  replaced by file_names

File: flow360/component/geometry.py (hunk 4 above)
Comment (feilin-flexcompute) [reply flow360/component/geometry.py:233]: fixed

**Review by maciej-flexcompute — APPROVED** [review]

**Pull request merged**